### DISCUSSION TITLE
Having trouble with linear synteny views

### DISCUSSION THREAD

**Discussion by jeremyjday** [GMOD/jbrowse-components#3716] [General]
Hello, 

I am having some trouble with the linear synteny view using a .chain file. I have attempted to load .chain files sourced from various places and have tried various species comparisons, but nothing shows up in the linear synteny view. Is there a requirement for a specific format for the chain file for this to work?

**Comment by cmdcolin** [comment]
Generally I have tested with chain files from UCSC. One thing is that it will load them all into memory up front so it can't be too large, but this is an example with a UCSC hg19 vs hg38 chain file https://jbrowse.org/code/jb2/v2.5.0/?config=test_data%2Fconfig_demo.json&session=share-_x8GHoVuZA&password=H0Wnw

**Reply by cmdcolin** [reply]
If you have an example of a chain file that doesn't load feel free to post, I can test it out (or can paste your config or commands you are using)

**Comment by jeremyjday** [comment]
Thank you for the reply. I am sourcing chain files from either UCSC or from Ensembl, and my main interest is in comparing the human genome sequence to rodents (mouse and rat). However, even when I try the hg19 to hg38 comparison, nothing loads in the space between linear genome tracks - it remains white unlike in your example. Is there an expectation for the .chain file to be unzipped? Do I need to load both genomes separately and prepare them in some way for the linear synteny view? I am supplying the file from my computer using the desktop version of JBrowse 2 on a Mac.

**Reply by cmdcolin** [reply]
just to be clear, do you mean the link at https://jbrowse.org/code/jb2/v2.5.0/?config=test_data%2Fconfig_demo.json&session=share-_x8GHoVuZA&password=H0Wnw is blank or when you load the chain file yourself manually?

## hg19 vs hg38 
here is a video showing the loading of hg19 vs hg38 with jbrowse desktop from scratch. note that the order of the assemblies is important in the import form, JBrowse currently cannot determine automatically which is the 'target' and 'query' assembly based on e.g. ref names alone (since both assemblies can use chr1-N as the naming scheme for example), so you have to manually specify them in the right order (the left and right side) on the linear synteny view start screen. this has been a persistent source of confusion, it may be possible for us to help by adding a sanity check based on reference sequence length to help this. in any case, if one is blank, try flipping the order of the assembly names in the linear synteny view import form (assuming you are using that)

[Screencast from 2023-05-22 09-48-08.webm](https://github.com/GMOD/jbrowse-components/assets/6511937/d111a43d-845c-4cbc-9b76-288027648656)

## human vs mouse (hs1 vs mm39)

note that human vs mouse actually can stretch the limits of loading chain files as synteny, as it too needs to be loaded into memory.  I made a demo using the 68 Mb file at https://hgdownload.soe.ucsc.edu/goldenPath/hs1/liftOver/hs1ToMm39.over.chain.gz for example unzips to 200Mb so it takes a little while to both download the 68Mb, then to unzip and parse the 200Mb of data, and render it. It is also a bit visually chaotic in some cases. here is a demo https://jbrowse.org/code/jb2/v2.5.0/?config=test_data%2Fhuman_vs_mouse.json&session=share-a-0dRFkh4H&password=QAbVu

**Reply by cmdcolin** [reply]
note that generally "orthologous gene" datasets are quite a bit smaller generally, so you could consider using those, but we are continuing to try to improve performance for whole genome alignment type data too

**Comment by jeremyjday** [comment]
Thanks for the reply. I am not sure how, but I finally got it to work using a .2bit assembly for the rat genome instead of a FASTA format. It might have also mattered that I supplied the chromAlias file when generating the rat assembly, but I cannot be sure.
